src/commands/deploy/deploy.ts

```diff
@@ -150,7 +147,7 @@ export async function deploy(context: IDeployWizardContext, confirmDeployment: b
                 window.showTextDocument(doc);
             }
 
-            await workspaceConfig.update(constants.configurationSettings.defaultWebAppToDeploy, '', context.configurationTarget);
+            await updateWorkspaceSetting(constants.configurationSettings.defaultWebAppToDeploy, '', context.fsPath);
```

Comment:
I think there's a bug with this one. Passing `context.configurationTarget` is def different than fspath

Reply:
The configurationTarget was to determine whether or not it was workspace or global specific.  With updateWorkspaceSetting, if undefined is passed for the fsPath, it will default to global effectively making the configurationTarget pointless.